injector/network_disruption.go

```diff
@@ -891,17 +912,110 @@ func (i *networkDisruptionInjector) handleFiltersForServices(interfaces []string
 	return nil
 }
 
+// handleFiltersForServices creates tc filters on given interfaces for hosts in disruption spec classifying matching packets in the given flowid
+func (i *networkDisruptionInjector) handleFiltersForHosts(interfaces []string, flowid string) error {
+	hosts := hostsWatcher{}
+
+	hostFilterMap, err := i.addFiltersForHosts(interfaces, i.spec.Hosts, flowid)
+	if err != nil {
+		return err
+	}
+
+	hosts.hostFilterMap = hostFilterMap
+
+	go i.watchHostChanges(interfaces, hosts, flowid)
```

Comment:
You might want to have a context or a channel to stop this goroutine in case of reinject.
You may not want several goroutine trying to apply filters concurrently.